activesupport/lib/active_support/test_case.rb

```diff
@@ -88,9 +89,7 @@ def parallelize(workers: :number_of_processors, with: :processes)
 
         self.lock_threads = false if defined?(self.lock_threads) && with == :threads
 
-        Minitest.parallel_executor = executor
-
-        parallelize_me!
+        Minitest.parallel_executor = Testing::ParallelizeWhenSuitableExecutor.new(executor)
```

Comment:
Do we have access to runnables before this? I think it makes more sense to not even create the executor if we don't have enough tests. Then the check could be moved up around line 79 where we return if workers are less than or equal to 1. I don't love the `WhenSuitableExecutor` name either and we could avoid renaming it by moving the check.

Reply:
@eileencodes yes that was my initial idea but, unfortunately, when `parallelize` runs, test cases aren't loaded yet, so you can't really count tests and make decisions. The executor path is not ideal, but we need some kind of lazy mechanism invoked from `Minitest.run`. I didn't want to extend/modify Minitest for this, so ended up with this approach as the best one I could find.

Happy to rename `ParallelizeWhenSuitableExecutor`. `ConditionalParallelExecutor` or something like that?

Reply:
I added some code to show a trace line when running tests with info on how they ran:

When in parallel:

```
Running 2829 tests in parallel in 8 processes
```

When not in parallel:

```
Running 15 tests in a single process (parallelization threshold is 20)
```